docs/reference/esql/esql-get-started.asciidoc

```diff
@@ -1,34 +1,434 @@
+calculate the median duration per client IP:
+
+[source,esql]
+----
+FROM sample_data
+| STATS median_duration = MEDIAN(event.duration) BY client.ip
+----
+
+[discrete]
+[[esql-getting-started-histogram]]
+=== Create a histogram
+
+To track statistics over time, {esql} enables you to create histograms using the
+<<esql-auto_bucket>> function. `AUTO_BUCKET` creates human-friendly buckets and
+returns a value for each row that corresponds to the resulting bucket the row
+falls into. 
+
+For example, to create hourly buckets for the data on October 23rd:
+
+[source,esql]
+----
+FROM sample_data
+| KEEP @timestamp
+| EVAL bucket = AUTO_BUCKET (@timestamp, 24, "2023-10-23T00:00:00Z", "2023-10-23T23:59:59Z")
+----
+
+Combine `AUTO_BUCKET` with <<esql-stats-by>> to create a histogram. For example,
+to count the number of events per hour:
+
+[source,esql]
+----
+FROM sample_data
+| KEEP @timestamp, event.duration
+| EVAL bucket = AUTO_BUCKET (@timestamp, 24, "2023-10-23T00:00:00Z", "2023-10-23T23:59:59Z")
+| STATS COUNT(*) BY bucket
+----
+
+Or the median duration per hour:
+
+[source,esql]
+----
+FROM sample_data
+| KEEP @timestamp, event.duration
+| EVAL bucket = AUTO_BUCKET (@timestamp, 24, "2023-10-23T00:00:00Z", "2023-10-23T23:59:59Z")
+| STATS median_duration = MEDIAN(event.duration) BY bucket
+----
+
+[discrete]
+[[esql-getting-started-enrich]]
+=== Enrich data
+
+{esql} enables you to <<esql-enrich-data,enrich>> a table with data from indices
+in {es}, using the <<esql-enrich>> command.
+
+image::images/esql/esql-enrich.png[align="center"]
+
+Before you can use `ENRICH`, you first need to
+<<esql-create-enrich-policy,create>> and <<esql-execute-enrich-policy,execute>>
+an <<esql-enrich-policy,enrich policy>>. The following requests create and
+execute a policy that links an IP address to an environment ("Development",
+"QA", or "Production"):
+
+[source,console]
+----
+PUT clientips
+{
+  "mappings": {
+    "properties": {
+      "client.ip": {
+        "type": "keyword"
+      },
+      "env": {
+        "type": "keyword"
+      }
+    }
+  }
+}
+
+PUT clientips/_bulk
+{ "index" : {}}
+{ "client.ip": "172.21.0.5", "env": "Development" }
+{ "index" : {}}
+{ "client.ip": "172.21.2.113", "env": "QA" }
+{ "index" : {}}
+{ "client.ip": "172.21.2.162", "env": "QA" }
+{ "index" : {}}
+{ "client.ip": "172.21.3.15", "env": "Production" }
+{ "index" : {}}
+{ "client.ip": "172.21.3.16", "env": "Production" }
+
+PUT /_enrich/policy/clientip_policy
+{
+  "match": {
+    "indices": "clientips",
+    "match_field": "client.ip",
+    "enrich_fields": ["env"]
+  }
+}
+
+PUT /_enrich/policy/clientip_policy/_execute
+----
+
+////
+[source,console]
+----
+DELETE /_enrich/policy/clientip_policy
+----
+// TEST[continued]
+////
+
+After creating and executing a policy, you can use it with the `ENRICH`
+command:
+
+[source,esql]
+----
+FROM sample_data
+| KEEP @timestamp, client.ip, event.duration
+| EVAL client.ip = TO_STRING(client.ip)
+| ENRICH clientip_policy
```

Comment:
I'd use the extended ENRICH syntax here, ie `ENRICH policy ON field`, that IMHO is more clear